folio-spring-system-user/src/main/java/org/folio/spring/service/SystemUserService.java

```diff
@@ -129,12 +141,36 @@ private UserToken getTokenLegacy(SystemUser user) {
         log.info("Login with legacy end-point not found");
         throw new AuthorizationException(CANNOT_RETRIEVE_OKAPI_TOKEN_FOR_TENANT + user.username());
       } else {
-        log.info("Login with legacy end-point returned unexpected error");
+        log.info(LOGIN_WITH_LEGACY_END_POINT_RETURNED_UNEXPECTED_ERROR);
         throw new AuthorizationException(CANNOT_RETRIEVE_OKAPI_TOKEN_FOR_TENANT + user.username());
       }
     }
   }
 
+  private UserToken getTokenLegacy(String tenantId, String username, String password) {
+    log.info("Attempting to issue token for system user [tenantId={}]", tenantId);
+    try {
+      var response =
+          authnClient.login(new UserCredentials(username, password));
+
+      var accessToken = response.getHeaders().get(X_OKAPI_TOKEN).get(0);
+
+      if (isNull(accessToken)) {
+        throw new AuthorizationException(CANNOT_RETRIEVE_OKAPI_TOKEN_FOR_TENANT + username);
+      }
+
+      return UserToken.builder().accessToken(accessToken).accessTokenExpiration(Instant.MAX).build();
+    } catch (FeignException fex) {
+      if (fex.status() == HttpStatus.NOT_FOUND.value()) {
+        log.info("Login with legacy end-point not found");
+        throw new AuthorizationException(CANNOT_RETRIEVE_OKAPI_TOKEN_FOR_TENANT + username);
+      } else {
+        log.info(LOGIN_WITH_LEGACY_END_POINT_RETURNED_UNEXPECTED_ERROR);
+        throw new AuthorizationException(CANNOT_RETRIEVE_OKAPI_TOKEN_FOR_TENANT + username);
+      }
+    }
+  }
+
```

Comment:
More code that already exits on the master branch as part of the previous merge.

I realize that you have changes here that are new, but the code that is already on master should not be included here.

Reply:
I can see that the method signature is different, but the body the method is now duplicated code.